lib/core/versions/latest/TransactionProcessor.ts

```diff
@@ -21,139 +21,122 @@ export default class TransactionProcessor implements ITransactionProcessor {
+
+      // If the code reaches here, it means that the batch of operations is valid, process the operations.
+      const endTimer = timeSpan();
+      await this.operationStore.put(operations);
+      console.info(`Processed batch '${anchorFile.batchFileHash}' of ${operations.length} operations. Time taken: ${endTimer.rounded()} ms.`);
+
+      return true;
+    } catch (error) {
+      if (error instanceof SidetreeError) {
+        // If error is potentially related to CAS network connectivity issues, we need to return false to retry later.
+        if (error.code === ErrorCode.CasNotReachable ||
+            error.code === ErrorCode.CasFileNotFound) {
+          return false;
+        }
+
+        console.info(`Ignoring error: ${error.message}`);
         return true;
       } else {
-        console.error(`Unexpected error processing anchor string, MUST investigate and fix: ${message}`);
+        console.error(`Unexpected error processing anchor string, MUST investigate and fix: ${error.message}`);
         return false;
       }
     }
+  }
 
-    const anchorFileHash = anchoredData.anchorFileHash;
-
+  private async downloadAndVerifyAnchorFile (anchorFileHash: string): Promise<AnchorFileModel> {
     console.info(`Downloading anchor file '${anchorFileHash}', max size limit ${ProtocolParameters.maxAnchorFileSizeInBytes} bytes...`);
     const anchorFileFetchResult = await this.downloadManager.download(anchorFileHash, ProtocolParameters.maxAnchorFileSizeInBytes);
 
     // No thing to process if the file hash is invalid. No retry needed.
     if (anchorFileFetchResult.code === FetchResultCode.InvalidHash) {
-      console.info(`Anchor file '${anchorFileHash}' is not a valid hash.`);
-      return true;
+      throw new SidetreeError(ErrorCode.AnchorFileHashNotValid, `Anchor file '${anchorFileHash}' is not a valid hash.`);
     }
 
     // No thing to process if the file size exceeds protocol specified size limit, no retry needed either.
     if (anchorFileFetchResult.code === FetchResultCode.MaxSizeExceeded) {
-      console.info(`Anchor file '${anchorFileHash}' exceeded max size limit ${ProtocolParameters.maxAnchorFileSizeInBytes} bytes.`);
-      return true;
+      throw new SidetreeError(
+        ErrorCode.AnchorFileTooLarge,
+        `Anchor file '${anchorFileHash}' exceeded max size limit ${ProtocolParameters.maxAnchorFileSizeInBytes} bytes.`
+      );
     }
 
     // Content for hash exists but is not a file. No retry needed.
     if (anchorFileFetchResult.code === FetchResultCode.NotAFile) {
-      console.info(`Anchor file hash '${anchorFileHash}' points to a content that is not a file.`);
-      return true;
+      throw new SidetreeError(ErrorCode.AnchorFileNotAFile, `Anchor file hash '${anchorFileHash}' points to a content that is not a file.`);
     }
 
     // If Content Addressable Storage is not reachable, mark the transaction for retry later.
```

Comment:
Nit: Update the comment as we're not 'marking' anything anymore (similar note for the other comments in this function).